src/python/pants/java/util.py

```diff
@@ -41,7 +41,8 @@ def _get_runner(classpath, main, jvm_options, args, executor,
 def execute_java(classpath, main, jvm_options=None, args=None, executor=None,
```

Comment:
Having the return type change based on the value of an optional argument is asking for trouble. How about creating a new method, that returns both, and then calling that from this method, discarding the workunit argument?